.github/workflows/storybook-deploy.yml

```
@@ -0,0 +1,29 @@
+name: Deploy Storybook to GitHub Pages
+on: [push]
+jobs:
+  publish:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - uses: pnpm/action-setup@v3
+        name: Install pnpm
+        with:
+          version: 8
+          run_install: true
+
+      - name: Build Storybook
+        run: pnpm run build-storybook
+
+      - name: Deploy to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@4.1.5
```

```suggestion
        uses: JamesIves/github-pages-deploy-action@v4.5.0
```
This _should_ resolve the branch issue.